api/src/main/java/org/apache/iceberg/util/StructProjection.java

```diff
@@ -39,9 +39,32 @@
    */
   public static StructProjection create(Schema schema, Set<Integer> ids) {
     StructType structType = schema.asStruct();
+    return create(structType, ids);
+  }
+
+  /**
+   * Creates a projecting wrapper for {@link StructLike} rows.
+   * <p>
+   * This projection does not work with repeated types like lists and maps.
+   *
+   * @param structType struct type of rows wrapped by this projection
+   * @param ids field ids from the row schema to project
+   * @return a wrapper to project rows
+   */
+  public static StructProjection create(StructType structType, Set<Integer> ids) {
     return new StructProjection(structType, TypeUtil.select(structType, ids));
   }
 
+  /**
+   * Creates a projecting wrapper for {@link StructLike} rows by copying from `other`.
+   *
+   * @param other the projection wrapper to copy from
+   * @return a wrapper to project rows
+   */
+  public static StructProjection create(StructProjection other) {
```

Comment:
Nit:  Instead of introducing a static `create` here , I'd prefer to introduce a new `copy` in `StructProjection`,  just similar to the `PartitionKey#copy`.   The name `copy` express the meaning that we are cloning the contents from the source object.

Reply:
I'm fine with either. I chose to follow the `create`-style because it might make it easier to find by looking at all the overloaded `create` methods.